todo_app/app.py

```diff
@@ -1,9 +1,42 @@
 from todo_app.data.session_items import delete_item
 from flask import Flask, render_template, request, url_for, redirect, session
 import requests
+import os
+import flask_login
+from flask_login import login_required, login_user, UserMixin, current_user, logout_user
+from oauthlib.oauth2 import WebApplicationClient
 from todo_app.flask_config import Config
 from todo_app.data import session_items as si
-from todo_app.classes import ViewModel
+from todo_app.classes import ViewModel, User
+from functools import wraps
+
+### Role decorators
+# Reader
+def reader_required(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        if current_user.is_anonymous == False:
+            if current_user.role == "reader" or current_user.role == "writer":
+                return f(*args, **kwargs)
+            else:
+                return redirect(url_for('index'))
+        else:
+            return f(*args, **kwargs)
+    return wrap
+
+# Writer
+def writer_required(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
```

Comment:
You could put @login_required on here so your routes don't need both `@login_required` and `@writer_required` each time.